Engine/ac/audiochannel.cpp

```diff
@@ -28,11 +28,15 @@ extern CCAudioClip ccDynamicAudioClip;
 
 int AudioChannel_GetID(ScriptAudioChannel *channel)
 {
+    std::lock_guard<std::recursive_timed_mutex> lock(channels_mutex);
+
     return channel->id;
```

Comment:
A small note, regarding this, I think the lock is redundant here, because ScriptAudioChannel object itself is never deleted during game play, and it's ID is always fixed.

I think there's a confusion between `ScriptAudioChannel *channel` and "channel" from `channels` array, which is actually SOUNDCLIP* (and may be deleted, and changed).

Reply:
Good point. I was using the locks to also mark out any audio usage but in this case it's not relevant.